### PR TITLE
feat: For steps with example have column names for parameters

### PR DESCRIPTION
**Pull request by lsaudon**
- [x] All tests passed

Add the possibility of having the name of the column rather than `param1` in the case of a step with examples.

Is it possible that in `<>` there are only numbers (like `<12.3>`)?

```feature
Feature: Testing feature
  Scenario: eating
    Given there are <number> cucumbers

    Examples:
      | number |
      |   12   |
      |   20   |
```
Before:
```dart
import 'package:flutter_test/flutter_test.dart';

Future<void> thereAreCucumbers(WidgetTester tester, dynamic param1) async {
  throw UnimplementedError();
}
```

After:
```dart
import 'package:flutter_test/flutter_test.dart';

Future<void> thereAreCucumbers(WidgetTester tester, dynamic number) async {
  throw UnimplementedError();
}
```

### CODE REVIEW

**Comment by olexale** [comment]
Thanks for the PR! I'll review it on the weekend.

File: lib/src/step/generic_step.dart [hunk 1]
```diff
@@ -19,31 +20,22 @@ class GenericStep implements BddStep {
...
     final params = parseRawStepLine(stepLine).skip(1);
-    if (params.isEmpty) {
-      final examples = examplesRegExp.allMatches(stepLine);
-      if (examples.isEmpty) {
-        return 'Future<void> $methodName($testerType $testerName) async';
-      } else {
-        return _generateSignature(examples.length, testerName);
-      }
+    if (params.isNotEmpty) {
+      return params.mapIndexed((p, index) => ', dynamic param${p + 1}').join();
```
Comment (olexale) [inline lib/src/step/generic_step.dart:31]: It is really a nice improvement for the scenario outline case, and as we are changing the generic step generation, I propose to make life slightly better for regular steps too. For scenario outlines we may detect the name of a parameter, and for other cases we may detect the type like that:
```dart
      return params
          .mapIndexed(
            (index, p) => ', ${_getGenericParameterType(p)} param${index + 1}',
          )
          .join();

...

  String _getGenericParameterType(String parameter) {
    if (parameter == 'true' || parameter == 'false') {
      return 'bool';
    }
    if (num.tryParse(parameter) != null) {
      return 'num';
    }
    if (parameter.startsWith('"') && parameter.endsWith('"') ||
        (parameter.startsWith("'") && parameter.endsWith("'"))) {
      return 'String';
    }
    return 'dynamic';
  }
```

What do you think?

File: test/steps_test.dart [hunk 2]
```diff
@@ -9,4 +10,35 @@ void main() {
         FeatureFile(featureDir: '$path.feature', package: path, input: '');
     expect(feature.getStepFiles().length, 0);
   });
+
+  test('When step with examples then param take column name', () {
```
Comment (olexale) [inline test/steps_test.dart:14]: Isn't it a duplicate of `Scenario Outline Step` from `test/scenario_outline_step_test.dart`? If you agree — please remove this test.

**Review by olexale — COMMENTED** [review]
Very nice small UX improvement that many of the users will notice! Thank you so much for it. I'm happy to merge the PR and publish a new version as soon as we are done with comments.

**Comment by olexale** [comment]
Thanks again, @lsaudon! I'll publish a new version of the package with your changes soon.

**Comment by lsaudon** [comment]
> Thanks again, @lsaudon! I'll publish a new version of the package with your changes soon.

Thank you for this package and your responsiveness.

**Pull request merged**